bigquery/bqiface/bqiface_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bqiface
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/iterator"
+)
+
+func TestIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("integration tests skipped in short mode")
+	}
+	projectID := os.Getenv("BQIFACE_PROJECT")
+	if projectID == "" {
+		t.Skip("missing BQIFACE_PROJECT environment variable")
+	}
+
+	ctx := context.Background()
+	c, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	client := AdaptClient(c)
+	defer client.Close()
+
+	ds := client.Dataset(fmt.Sprintf("bqiface_%d", time.Now().Unix()))
+	var md DatasetMetadata
+	md.DefaultTableExpiration = time.Hour
+	var ae AccessEntry
+	ae.Role = bigquery.OwnerRole
+	ae.EntityType = bigquery.SpecialGroupEntity
+	ae.Entity = "projectOwners"
+	md.Access = []*AccessEntry{&ae}
+	err = ds.Create(ctx, &md)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	md2, err := ds.Metadata(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if got, want := md2.DefaultTableExpiration, md.DefaultTableExpiration; got != want {
+		t.Errorf("DefaultTableExpiration: got %s, want %s", got, want)
+	}
+	if got, want := len(md2.Access), 1; got != want {
```

Comment:
Would reflect.DeepEqual work here instead of len + element check?

(I legitimately don't recall)

Reply:
I'd like to discourage reflect.DeepEqual for these sorts of things. The cmp package is the right choice in general, but I don't want to make this too complicated since it's really just a smoke test.

Reply:
SGTM